Tests/UnitTests/Core/Surfaces/LineSurfaceTests.cpp

```diff
@@ -266,6 +275,59 @@ BOOST_AUTO_TEST_CASE(LineSurfaceTransformRoundTripEtaStability) {
   }
 }
 
+BOOST_AUTO_TEST_CASE(LineSurfaceIntersection) {
+  using namespace Acts::UnitLiterals;
+
+  auto eps = 1e-10;
+
+  auto direction = Vector3(1, 1, 100).normalized();
+  auto boundVector =
+      (BoundVector() << 1_cm, 1_cm, VectorHelpers::phi(direction),
+       VectorHelpers::theta(direction), 1, 0)
+          .finished();
+  auto pathLimit = 1_cm;
+
+  auto surface = std::make_shared<LineSurfaceStub>(Transform3::Identity());
+
+  auto initialParams = BoundTrackParameters(surface, boundVector, 1);
+
+  auto propagator = Propagator<StraightLineStepper>({});
+
+  auto displacedParameters =
+      CurvilinearTrackParameters(Vector4::Zero(), Vector3::Zero(), 1);
+  {
+    auto options = PropagatorOptions<>(tgContext, {});
+    options.direction = Acts::Direction::Backward;
+    options.pathLimit = pathLimit;
+
+    auto result = propagator.propagate(initialParams, options);
+    BOOST_CHECK(result.ok());
+    BOOST_CHECK(result.value().endParameters);
+
+    displacedParameters = result.value().endParameters.value();
+  }
+
+  auto intersection =
+      surface->intersect(tgContext, displacedParameters.position(tgContext),
+                         displacedParameters.unitDirection());
+  CHECK_CLOSE_ABS(intersection.intersection.pathLength, pathLimit, eps);
```

Comment:
So this takes parameters at `0,0,0`, straight line steps backwards `1cm`, then checks if it's intersecting with the surface at `1cm`? If so what's the point of the propagation? Can't you just literally move the track parameters position backwards by `1cm`?

Reply:
it takes the bound parameters we used before and steps away. we could do the math explicitly but then I have to do manual `localToGlobal` and add a displacement vector which seemed more work at the time. this way we also check the consistency of the straight line intersection and the straight line propagation

Reply:
ok